### PR TITLE
FISH-6711 FISH-6712 Upgrade OSGi Util Promise and Function

### PR DESCRIPTION
**Pull request by Pandrex247**
Port of https://github.com/payara/Payara/pull/6078

### CODE REVIEW

**Comment by Pandrex247** [comment]
Jenkins test please

File: nucleus/osgi-platforms/felix/src/main/resources/config/osgi.properties [hunk 1]
```diff
@@ -580,32 +580,77 @@ org.osgi.framework.system.capabilities= \
  ${eecap-${java.specification.version}}
```
Comment (pdudits) [inline nucleus/osgi-platforms/felix/src/main/resources/config/osgi.properties:580]: Thinking about it once again, I believe we need only this:

```suggestion
 osgi.ee;osgi.ee="JavaSE/compact1";version="1.8", \
 osgi.ee;osgi.ee="JavaSE/compact2";version="1.8",\
 osgi.ee;osgi.ee="JavaSE/compact3";version="1.8", ${eecap-${java.specification.version}}
```
Insteead of all changes below. Compact versions only existed in Java 1.8 and were superseded with Jigsaw. This should be enough to make osgi.function happy.

Let me quickly try that out...

Comment (pdudits) [reply nucleus/osgi-platforms/felix/src/main/resources/config/osgi.properties:581]: Yep that works well.

**Comment by pdudits** [comment]
Jenkins test please

**Review by pdudits — APPROVED** [review]

**Pull request merged**